lib/utils.sh

```diff
@@ -102,3 +103,29 @@ get_nodes() {
 get_pods() {
 	kubectl get pods --all-namespaces --no-headers
 }
+
+container_exists() {
```

Comment:
What's the benefit for adding such two new functions(run_container and container_exists) here?

Reply:
run_container is reused in both providers and runs  **container** passed. In the providers, I am using it to run registry. 

why are there 2 different functions is similar to any language these functions make the intent clear ... 
e.g 

```
until container_exits docker foobar; do 
  .. wait for it
done
```
is easier to understand IMHO than 
```
until $ctr_cmd ps -a | grep -q "foobar"; do 
done
```
which actually used to be 
```
until [ -z $( $ctr_cmd ps -a | grep -q "foobar") ] ; do 
done
```
in short if there is an expression that returns boolean and is reused elsewhere, make that a function.